src/components/ItemsList.js

```diff
@@ -43,41 +47,65 @@ const ItemsList = () => {
     history.push('/additem');
   };
 
+  const clearText = (event) => {
+    event.preventDefault();
+    setSearchTerm('');
+  };
 
   return (
     <div className="items-list">
       <h1>Your Shopping List</h1>
       {loading && <p>Loading...</p>}
       {error && <p>An error has occured...</p>}
       {shoppingList && !shoppingList.length && (
-            <div className="add-item">
-              <p>You haven't created a shopping list yet...</p>
-              <button type="submit" onClick={handleRedirect}>
-                Add First Item
-              </button>
-            </div>
-          )}
-      <form>
+        <div className="add-item">
+          <p>You haven't created a shopping list yet...</p>
+          <button type="submit" onClick={handleRedirect}>
+            Add First Item
+          </button>
+        </div>
+      )}
+      <form onReset={clearText} className="search-list">
```

Comment:
The form works perfectly here. I have one idea: 

I wanted to suggest maybe it is a good idea to create a new component `SearchBar` that will render the search bar? This way you can work more React way and delegate tasks by individual components. 
You can keep the state and logic in the `itemsList` component and pass down the state to the `SearchBar` component. This can also make the code more organized. 

If you decide to refactor the code, also keep in mind the semantic use of HTML. Maybe it will be better to achieve the same functionality using the `div` tag instead of the `form` tag. The idea here is that since we are not submitting any data (but just filtering), we could avoid using the `form` tag. You can read more about forms [here](https://reactjs.org/docs/forms.html) and [here](https://html.com/tags/form/).

Reply:
I changed the `form` tag to a `div` one. First, it seemed that I can use reset only within a `form` tag. But after refreshing the page everything worked out fine this time 👍🏽

Reply:
I like the idea of having more components to make our code more readable but for me, I would not know how to do that. How to get access to some data I would need for this particular component, etc. So I am not much of a help here!

Reply:
Ok or maybe I am lucky and at least for this component I got it to work (Would be too good to be true, so pls can anyone check)

Reply:
👏👏👏 This is amazing! You got it @YuriDevAT!

Reply:
There is one more detail I should mention. 

As the AC asks, we need to conditional render the "X" button. 
Just a hint: Maybe check if `searchTerm` has a value. If it does not, then don't render the button. Otherwise, render it. 
Please let me know if you have any questions 👍

Reply:
I think I did it 👀